general/mem_manager.hpp

```diff
@@ -573,6 +573,10 @@ class MemoryManager
    /// True if Configure() was called.
    static bool configured;
 
+   /** @brief If true, class Memory will register the base Memory when creating
+       aliases with Memory<T>::MakeAlias(). */
+   static bool register_alias_bases;
```

Comment:
Maybe develop a bit why it makes sense from a design point of view, it's not obvious to me to know why we would want to not register the bases. It seems to me that it's just a convenience variable that is false when CPU only, and true when there is a device, but has no design purpose.

Reply:
There's a bit more in the documentation of this variable's the access method, `RegisterAliasBases()`. I'll just point the comment here to that method and expand the comment there.

Reply:
Nevermind, the next suggestion supersedes this.